src/state/subscribers/reducers.js

```diff
@@ -0,0 +1,41 @@
+import {
+    REQUEST_FAILED,
+    SUBMIT_SUBSCRIBER_SUCCESS,
+    SUBMIT_SUBSCRIBER,
+    REQUEST_ALL_SUBSCRIBERS_SUCCESS,
+} from './constants';
+
+const initialState = {
+    error: null,
+    allSubscribers: [],
+};
+
+const subscriberReducer = (state = initialState, {type, payload}) => {
+    switch (type) {
+        case SUBMIT_SUBSCRIBER_SUCCESS:
+            const updateAllSubscribers = [...state.allSubscribers];
+            const index = state.allSubscribers.findIndex((el) => {
+                return el.key === payload.key;
+            });
+            if (index < 0) {
+                updateAllSubscribers.push(payload);
+            } else {
+                updateAllSubscribers[index] = payload;
+            }
+            return {
```

Comment:
shorter, using lodash.map: 
```
allSubscribers: map(state.allSubscribers, (person) => {
if (person.key === payload.key){
     return payload;
}
return person
})
```
or without lodash: 
```
allSubscribers: state.allSubscribers.map((person) => {
if (person.key === payload.key){
    return payload;
}
return person
})
```

Reply:
The problem is I need to add the subscriber if it doesn't already exist. I'm not in love with the code I have now & would love to simplify it, but if I make the change you suggest the submit will only work for updating, not creating.

Reply:
For the update verses new logic, that should be two different actions. UPDATE_SUBSCRIBER and ADD_NEW_SUBSCRIBER so that the logic within each switch option is clear and simple.

Reply:
Agreed that this is much better, thank you